models/base_model.py

```diff
@@ -886,14 +886,26 @@ def print_networks(self, verbose):
             verbose (bool) -- if verbose: print the network architecture
         """
         print("---------- Networks initialized -------------")
+        self.nets_arch = ""
         for name in self.model_names:
             if isinstance(name, str):
                 net = getattr(self, "net" + name)
                 num_params = 0
                 for param in net.parameters():
                     num_params += param.numel()
-                if verbose:
-                    print(net)
+                if self.niter == 0:
```

Comment:
This stuff should go in `visualizer.py` or related, not really in core model base code.